src/librbd/image/OpenRequest.cc

```diff
@@ -482,6 +482,7 @@ Context *OpenRequest<I>::handle_v2_get_data_pool(int *result) {
       }
       m_image_ctx->data_ctx.close();
     } else {
+      m_image_ctx->rebuild_data_io_context();
```

Comment:
@dillaman `rebuild_data_io_context` uses the current `data_ctx` namespace, but it is set below. Is it intentional?

Reply:
Negative -- will fix in next push